Automattic/wp-calypso
# Packages: Move build config to calypso-build package

First steps. See p9oQ9f-87-p2 for context. Somewhat similar to https://github.com/WordPress/gutenberg/pull/13814 -- ideally we'll converge at some point.

Jetpack counterpart PR: https://github.com/Automattic/jetpack/pull/11344

#### Changes proposed in this Pull Request

#### TODO:
- [ ] Make work for JP
- [ ] Separate package for style assets?
- [ ] Use for Calypso itself

#### Tasks we'll need to do to follow through with this

- [ ] Our (S)CSS buildchain. Depends on stuff in `assets/stylesheets/_shared`, which I'm currently directly including. Wondering if _that_ should be its own package that the `calypso-builder` package can depend on. Est: 3d
  - To make things more fun, this is [run](https://github.com/Automattic/wp-calypso/blob/master/bin/build-custom-properties-css.js) through `node-sass` during build (and our files subsequently depend on the resulting `public/custom-properties.css`). If we turn this into an npm, this would need to be part of the published package.
  - We could maybe drop the SASS compilation on this one and just turn it into CSS? It's mostly just custom properties definitions after all.
- [ ] Need to parametrize root dir in webpack config (instead of hard-wired `__dirname`). Est: .5d
- [ ] References to Calypso's `config`. E.g. used for env, project name,... Hopefully avoidable or parametrizable. Est: 1d
- [ ] Factor out a lot of Calypso specific config that doesn't carry over, e.g. server (section) loaders. Est: 2d
- [ ] 'Actual' code from Calypso (that we'd need to publish as npms -- regardless of which build tool we end up using in JP). Not a lot to see here:
  - `components/jetpack-logo`
  - `lib/format-currency/currencies`
  - `lib/route/path`
  - `lib/simple-payments/constants`
  - `lib/simple-payments/utils`
- Est: .5d if we copy/duplicate, 3d if we package
- [ ] Stuff relevant for `client/extensions`. Is any of that even still used? Can we yank it out altogether? (This is probably minor/optional)

#### Testing instructions

See https://github.com/Automattic/jetpack/pull/11344

/cc @Automattic/team-calypso @Automattic/gutenpack
